cmd/soroban-cli/src/config/identity/show.rs

```diff
@@ -0,0 +1,38 @@
+use crate::config::{locator, secret};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Config(#[from] locator::Error),
+
+    #[error(transparent)]
+    Secret(#[from] secret::Error),
+
+    #[error(transparent)]
+    StrKey(#[from] stellar_strkey::DecodeError),
+}
+
+#[derive(Debug, clap::Args)]
+pub struct Cmd {
+    /// Name of identity to lookup
+    pub name: String,
+
+    /// If identity is a seed phrase use this hd path, default is 0
+    #[clap(long)]
+    pub hd_path: Option<usize>,
+}
+
+impl Cmd {
+    pub fn run(&self) -> Result<(), Error> {
+        println!("{}", self.private_key()?.to_string());
+        Ok(())
+    }
+
+    pub fn private_key(&self) -> Result<stellar_strkey::ed25519::PrivateKey, Error> {
+        let res = locator::read_identity(&self.name)?;
+        let key = res.key_pair(self.hd_path)?;
+        Ok(stellar_strkey::ed25519::PrivateKey::from_payload(
+            key.public.as_bytes(),
+        )?)
```

Comment:
Does this work even if it was generated with a seed phrase?

Reply:
Yep! key_pair uses the hd_path if the identity was a seed phrase. This is also why the default is a seed phrase.